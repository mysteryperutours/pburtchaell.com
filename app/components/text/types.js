export const HEADER_1 = 'h1';
export const HEADER_2 = 'h2';
export const HEADER_3 = 'h3';
export const BODY = 'p';
