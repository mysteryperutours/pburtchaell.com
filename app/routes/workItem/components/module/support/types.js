const types: Object = {
  MODULE_MEDIA: 'module-media',
  MODULE_TEXT: 'module-text',
  MODULE_MEDIA_FULL_WIDTH: 'module-media-full-width',
  MODULE_MEDIA_LEFT: 'module-media-left',
  MODULE_MEDIA_RIGHT: 'module-media-right'
};

export default types;
