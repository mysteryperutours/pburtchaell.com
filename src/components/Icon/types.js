export const GITHUB = 'GITHUB';
export const TWITTER = 'TWITTER';
export const FACEBOOK = 'FACEBOOK';
export const DRIBBBLE = 'DRIBBBLE';
