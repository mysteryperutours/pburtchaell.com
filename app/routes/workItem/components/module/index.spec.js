import React from 'react';
import Module from './index';
import types from './support/types';

const MOCK_MODULE_MEDIA = {
  type: types.MODULE_MEDIA,
  children: [{
    type: 'image',
    meta: {
      src: 'https://path/to/image.png',
      alt: 'Foo',
      height: 500,
      width: 700
    }
  }]
};

const MOCK_MODULE_TEXT = {
  type: types.MODULE_TEXT,
  children: [{
    type: 'text',
    children: 'Foo'
  }]
};

describe(Module.name, () => {
  it('renders module with type media', () => {
    const component = (
      <Module
        type={MOCK_MODULE_MEDIA.type}
      >
        {MOCK_MODULE_MEDIA.children}
      </Module>
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  it('renders module with type text', () => {
    const component = (
      <Module
        type={MOCK_MODULE_TEXT.type}
      >
        {MOCK_MODULE_TEXT.children}
      </Module>
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  it('renders module with multiple children', () => {
    const component = (
      <Module
        type={MOCK_MODULE_TEXT.type}
      >
        {[MOCK_MODULE_TEXT.children[0], MOCK_MODULE_TEXT.children[0]]}
      </Module>
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
