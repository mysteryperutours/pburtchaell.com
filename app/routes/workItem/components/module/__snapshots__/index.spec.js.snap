exports[`Module renders module with multiple children 1`] = `
<Row
  className="case-study-module module-text"
  defaultColumn={true}
  elementType="section">
  <Column
    elementType="div"
    offset={2}
    size={8}
    style={Object {}}>
    <Text
      type="p">
      Foo
    </Text>
    <Text
      type="p">
      Foo
    </Text>
  </Column>
</Row>
`;

exports[`Module renders module with type media 1`] = `
<Row
  className="case-study-module module-media"
  defaultColumn={true}
  elementType="section">
  <Column
    elementType="div"
    offset={2}
    size={8}
    style={Object {}}>
    <Image
      alt="Foo"
      height={500}
      src="https://path/to/image.png"
      width={700} />
  </Column>
</Row>
`;

exports[`Module renders module with type text 1`] = `
<Row
  className="case-study-module module-text"
  defaultColumn={true}
  elementType="section">
  <Column
    elementType="div"
    offset={2}
    size={8}
    style={Object {}}>
    <Text
      type="p">
      Foo
    </Text>
  </Column>
</Row>
`;
